Fixes to S3 config in data prepper. (#6787)
opensearch-project/documentation-website: _data-prepper/pipelines/configuration/sinks/s3.md

          region: us-east-1
          sts_role_arn: arn:aws:iam::123456789012:role/Data-Prepper
        max_retries: 5
        bucket: bucket_name
        object_key:

>           path_prefix: my-elb/%{yyyy}/%{MM}/%{dd}/

Reviewer: I think we should stick with `my-logs` here. This example is not specifically for ELB logs.

>           path_prefix: my-elb/%{yyyy}/%{MM}/%{dd}/

Reviewer 2: ```suggestion
          path_prefix: my-logs/%{yyyy}/%{MM}/%{dd}/
```

Option | Required | Type | Description
:--- | :--- | :--- | :---
`bucket` | Yes | String | The object from which the data is retrieved and then stored. The `name` must match the name of your object store.
`codec` | Yes | [Codec](#codec) | The format of the output data.
`aws` | Yes | AWS | The AWS configuration. See [aws](#aws) for more information.
`threshold` | Yes | [Threshold](#threshold-configuration) | Configures when to write an object to S3. 

> `object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. File pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket.  `path_prefix` is configurable.

Author: So I couldn't tell if this was on purpose or not but checking the code the parameter is not configurable (even though the old docs made it look configurable).

> `object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. File pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket.  `path_prefix` is configurable.

Reviewer 2: ```suggestion
`object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. File pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket. The `path_prefix` is configurable.
```

Option | Required | Type | Description
:--- | :--- | :--- | :---

> `bucket` | Yes | String | The object from which the data is retrieved and then stored. The `name` must match the name of your object store.

Reviewer: What was the motivation for this rewording? The value provided here is the name of a bucket for writing.

In S3 terminology, the bucket is where objects are stored. So the "bucket" term is more accurate this this location. Perhaps:

> The name of the S3 bucket to which objects are stored

> `bucket` | Yes | String | The object from which the data is retrieved and then stored. The `name` must match the name of your object store.

Reviewer 2: ```suggestion
`bucket` | Yes | String | The name of the S3 bucket to which objects are stored. The `name` must match the name of your object store.
```

> `codec` | Yes | [Codec](#codec) | The format of the output data.

Reviewer: I like this change overall. But, I also think it should clearly connect to codecs since it has to be an existing Data Prepper codec.

> The codec determining the format of output data

> `codec` | Yes | [Codec](#codec) | The format of the output data.

Reviewer 2: ```suggestion
`codec` | Yes | [Codec](#codec) | The codec determining the format of output data.
```

`aws` | Yes | AWS | The AWS configuration. See [aws](#aws) for more information.
`threshold` | Yes | [Threshold](#threshold-configuration) | Configures when to write an object to S3. 

> `object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. File pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket. The `path_prefix` is configurable.

Reviewer 2: ```suggestion
`object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. The file pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket. The `path_prefix` is configurable.
```

> `object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. The file pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}` found inside the root directory of the bucket. The `path_prefix` is configurable.

Reviewer: ```suggestion
`object_key` | No | Sets the `path_prefix` and the `file_pattern` of the object store. The file pattern is always `events-%{yyyy-MM-dd'T'hh-mm-ss}`. By default, those objects are found inside the root directory of the bucket. The `path_prefix` is configurable.
```

Using the `path_prefix` will make the objects show up in somewhere that is not a "root" directory. So I tried to clarify that.
